orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStage.groovy

```diff
@@ -80,31 +85,8 @@ class BakeStage implements StageDefinitionBuilder {
     deployRegions.addAll(stage.context.regions as Set<String> ?: [])
 
     if (!deployRegions.contains("global")) {
-      deployRegions.addAll(stage.execution.stages.findAll {
-        it.type == "deploy"
-      }.collect {
-        Set<String> regions = it.context?.clusters?.inject([] as Set<String>) { Set<String> accum, Map cluster ->
-          if (cluster.cloudProvider == stage.context.cloudProviderType) {
-            accum.addAll(cluster.availabilityZones?.keySet() ?: [])
-          }
-          return accum
-        } ?: []
-        if (it.context?.cluster?.cloudProvider == stage.context.cloudProviderType) {
-          regions.addAll(it.context?.cluster?.availabilityZones?.keySet() ?: [])
-        }
-        return regions
-      }.flatten())
+      deployRegions.addAll(regionCollector.getRegionsFromDescendentStages(stage))
```

Comment:
I dig it!

Reply:
thanks for helping me arrive at this solution!